### ISSUE TITLE
Allow excluding class expressions in max-classes-per-file

### ISSUE THREAD

**Issue by carlo-quinonez** [eslint/eslint#14855]
**What rule do you want to change?**
Add an option to max-classes-per-file to no longer count class expressions.

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option. Say, "allowClassExpressions"?

**Please provide some example code that this change will affect:**

```js
class VisitorFactory {
    forDescriptor(descriptor) {
        return class {
            visit(node) {
                return `Visiting ${descriptor}.`;
            }
        };
    }
}

```

**What does the rule currently do for this code?**
Counts both class declarations and expressions no matter what.

**What will the rule do after it's changed?**
If this new option is enabled, only consider class declarations.

**Are you willing to submit a pull request to implement this change?**
Possibly, but this was [proposed earlier](https://github.com/eslint/eslint/issues/11883) and the OP @JoshuaKGoldberg offered to submit a PR for this.

All of this text was copied from the [original issue](https://github.com/eslint/eslint/issues/11883). There was a [discussion](https://github.com/eslint/eslint/discussions/14684) about this and @nzakas suggested I open a new issue.

**Comment by JoshuaKGoldberg** [comment]
I'm still willing & able to send a PR - just got confused by the process!

**Comment by nzakas** [comment]
I think this makes sense. Let’s just get a couple other folks on the team to chime in.

**Comment by btmills** [comment]
I can see only wanting to count class declarations against the limit :+1:

**Comment by nzakas** [comment]
Marking as accepted.
